graphstorm-processing/tests/test_repartition_files.py

```diff
@@ -0,0 +1,283 @@
+    metadata_path = os.path.join(_ROOT, "resources/repartitioning/partitioned_metadata.json")
+    shutil.copyfile(metadata_path, os.path.join(TEMP_DATA_PREFIX, "partitioned_metadata.json"))
+
+    # The edge structure files will have different row count from all data files.
+    edges_rows = [8, 8, 8, 8, 18]
+    # For edge data files, the label will have a different row count from the rest
+    # and [10, 10, 10, 10, 10] will be the most frequent row count, so all other
+    # files will be re-partitioned to match that.
+    per_file_rows = [
+        [10, 12, 10, 8, 10],
+        [10, 10, 10, 10, 10],
+        [10, 10, 10, 10, 10],
+    ]
+
+    col_names = [
+        "label",
+        "feature_one",
+        "feature_two",
+    ]
+
+    # Read the metadata to ensure our generated files will match the expected counts
+    with open(
+        os.path.join(TEMP_DATA_PREFIX, "partitioned_metadata.json"), "r", encoding="utf-8"
+    ) as metafile:
+        metadata_dict = json.load(metafile)
+    # Check metadata file content against the row counts defined in this function
+    for meta_etype in metadata_dict["edge_type"]:
+        edge_structure_counts = metadata_dict["edges"][meta_etype]["row_counts"]
+        assert edge_structure_counts == edges_rows
+        for col_name, generated_rows in zip(col_names, per_file_rows):
+            edge_data_counts = metadata_dict["edge_data"][meta_etype][col_name]["row_counts"]
+            assert edge_data_counts == generated_rows
+
+    # Generate data and write edge data files to disk
+    for row_distribution, col_name in zip(per_file_rows, col_names):
+        total_rows = 0
+        feat_path = os.path.join(
+            TEMP_DATA_PREFIX,
+            "edge_data",
+            f"dummy_type-{col_name}",
+            "parquet",
+        )
+        os.makedirs(feat_path)
+        for i, row_count in enumerate(row_distribution):
+            feat_part = create_feature_table(col_name, row_count, total_rows)
+            filename = f"part-{str(i).zfill(5)}.parquet"
+            pq.write_table(feat_part, os.path.join(feat_path, filename))
+            total_rows += row_count
+
+    # Generate edge structure files
+    # The edges will have src_int_ids from 0 to 49, and dst_ids from 50 to 99
+    src_col: pa.Array = pa.array(range(0, 50))
+    dst_col: pa.Array = pa.array(range(50, 100))
+    edges_table = pa.table([src_col, dst_col], names=["src_int_id", "dst_int_id"])
+
+    edges_path = os.path.join(TEMP_DATA_PREFIX, "edges", "dummy_type", "parquet")
+    os.makedirs(edges_path)
+    total_edges = 0
+    for i, edge_rows in enumerate(edges_rows):
+        filename = f"part-{str(i).zfill(5)}.parquet"
+        pq.write_table(
+            edges_table[total_edges : total_edges + edge_rows], os.path.join(edges_path, filename)
+        )
+        total_edges += edge_rows
+
+    yield TEMP_DATA_PREFIX
+
+    # Cleanup all temp files after the entire module has been tested
+    shutil.rmtree(TEMP_DATA_PREFIX)
+
+
```

Comment:
Can you add tests for collect_frequencies_for_data_counts and verify_metadata?

Reply:
Those are also covered in the main function tests, do we want to have separate unit tests for them? In the sense that they're not meant to be called externally, we can rely on hiding their implementation and avoid testing them (thereby adding maintenance if we decide to change their implementation, e.g. by switching from most frequent to largest file size).